impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaListener.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.testing.kafka.common;
+
+import java.util.Objects;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * The type Kafka Listener.
+ *
+ * @param bind       the bind address
+ * @param kafkaNet   the connect address within the kafka network
+ * @param advertised the connect address from the client network
+ */
+public record KafkaListener(@NonNull KafkaEndpoint bind, @NonNull KafkaEndpoint kafkaNet, @NonNull KafkaEndpoint advertised) {
```

Comment:
I think I'd prefer to see the `*Address` methods restored here to provide the correctly formatted address patterns.